### PR TITLE
Add route for /mux-player

### PR DESCRIPTION
**Pull request by dylanjha**
✅  add route for `/v/:playbackId/mux-player`
✅  show an error message for player-specific routes that don't exist
✅  refactor routing based on review feedback: https://github.com/muxinc/stream.new/pull/91#discussion_r812157054

# Blockers before 🚢 

- [ ] code review / qa
- [ ] need a more recent alpha build of mux-player-react (the recent Mux Data metadata change is the blocker)
- [x] maybe want to pull in @davekiss changes in #89 and commence an a/b test on the root playback page (either do it as part of this PR, or do a followup)

### CODE REVIEW

File: components/player-page.tsx [hunk 1]
```diff
@@ -44,6 +44,12 @@ const Playback: React.FC<PageProps> = ({ playbackId, videoExists, shareUrl, post
     }
   }, [videoExists]);
 
+  useEffect(() => {
+    if (playerType && !VALID_PLAYER_TYPES.includes(playerType)) {
+      setErrorMessage(`Don't know how to load the player called: ${playerType}`);
+    }
+  }, [playerType]);
```
Comment (dylanjha) [inline components/player-page.tsx:51]: Looks like:

error-player_2022-02-22_22-29-55

File: constants.js [hunk 2]
```diff
@@ -2,3 +2,6 @@ export const MUX_HOME_PAGE_URL = 'https://mux.com?utm_source=stream-new';
 export const MUX_TERMS_URL = 'https://mux.com/terms?utm_source=stream-new';
 export const OPEN_SOURCE_URL = 'https://github.com/muxinc/stream.new';
 export const HOST_URL = 'https://stream.new';
+export const PLYR_TYPE = "plyr";
+export const MUX_PLAYER_TYPE = "mux-player";
```
Comment (dylanjha) [inline constants.js:6]: @davekiss I'm seeing a few places where `plyr` and `mux-player` is hard coded, maybe want to use these constants instead?

File: pages/v/[id]/_middleware.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+
+const COOKIE_KEY = 'streamPlayerType';
+const SPLIT_PERCENTAGE = 0.1;
```
Comment (dylanjha) [inline pages/v/[id]/_middleware.ts:5]: Lets start at 0.2 YOLO.

Also could move this to an env var. Env vars are accessible in Next middleware, right?

File: components/mux-player.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+};
+
+const MuxPlayerInternal: React.FC<Props> = ({ playbackId, poster, currentTime, onLoaded, onError }) => {
+  useEffect(() => {
+    onLoaded();
+  }, []);
+
+  return (
+    <>
+      <MuxPlayer playbackId={playbackId} poster={poster} startTime={currentTime} onError={(err) => onError(err as ErrorEvent)} envKey={process.env.NEXT_PUBLIC_MUX_ENV_KEY} streamType="on-demand" />
```
Comment (cjpillsbury) [inline components/mux-player.tsx:19]: Did you do a simple smoke test on `onError`?

File: components/mux-player.tsx [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  onError: (error: ErrorEvent) => void;
+};
+
+const MuxPlayerInternal: React.FC<Props> = ({ playbackId, poster, currentTime, onLoaded, onError }) => {
+  useEffect(() => {
+    onLoaded();
+  }, []);
+
+  return (
+    <>
```
Comment (cjpillsbury) [inline components/mux-player.tsx:18]: nit: drop the fragment if not needed.

File: lib/player-page-utils.ts [hunk 6]
```diff
@@ -8,6 +8,7 @@ export type Props = {
   poster: string;
   aspectRatio?: number;
   videoExists: boolean;
+  playerType?: | 'plyr' | 'mux-player';
```
Comment (cjpillsbury) [inline lib/player-page-utils.ts:11]: nit: this should use `PlayerTypes` in `constants`, above, correct?

File: package.json [hunk 7]
```diff
@@ -13,14 +13,16 @@
...
     "@mux/mux-node": "^3.2",
     "@mux/upchunk": "^2.2",
     "@types/probe-image-size": "^7.0.0",
     "copy-to-clipboard": "^3.3.1",
     "got": "^11.8.3",
-    "hls.js": "^1.0.0",
+    "hls.js": "^1.1.5",
+    "js-cookie": "^3.0.1",
     "micro": "9.3.5-canary.3",
-    "mux-embed": "^3.3.0",
```
Comment (cjpillsbury) [inline package.json:23]: 🤯 that out of date??

File: pages/v/[id]/_middleware.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import { NextResponse } from 'next/server';
+import type { NextRequest } from 'next/server';
+import { PLYR_TYPE, MUX_PLAYER_TYPE } from '../../../constants';
+
+const COOKIE_KEY = 'streamPlayerType';
+const SPLIT_PERCENTAGE = +(process.env.PLAYER_SPLIT_TESTING_PERCENTAGE || 0.2);
+
+export function middleware(request: NextRequest): NextResponse {
```
Comment (cjpillsbury) [inline pages/v/[id]/_middleware.ts:8]: 👌

**Review by cjpillsbury — APPROVED** [review]
A couple of non-blocking nits. Defer to you on prioritizing.

File: components/mux-player.tsx (hunk 4 above)
Comment (dylanjha) [reply components/mux-player.tsx:19]: discussed in slack

**Pull request merged**